cli/src/pcluster/models/imagebuilder.py

```diff
@@ -12,160 +12,112 @@
-        self.distribution_configuration = distribution_configuration
-        self.terminate_instance_on_failure = Resource.init_param(terminate_instance_on_failure, default=True)
-
-
-# ---------------------- ImageBuilder ---------------------- #
-
-
-class ImageBuilder(Resource):
-    """Represent the configuration of an ImageBuilder."""
-
-    def __init__(
-        self,
-        image: Image,
-        build: Build,
-        dev_settings: ImagebuilderDevSettings = None,
-    ):
-        super().__init__()
-        self.image = image
-        self.build = build
-        self.dev_settings = dev_settings
-
-    def _validate(self):
-        self._execute_validator(
-            EbsVolumeTypeSizeValidator,
-            volume_type=ROOT_VOLUME_TYPE,
-            volume_size=self.image.root_volume.size,
-        )
-        self._execute_validator(
-            AMIVolumeSizeValidator,
-            volume_size=self.image.root_volume.size,
-            image=self.build.parent_image,
-        )
+import json
+import logging
+
+from common.aws.aws_api import AWSApi
+from common.aws.aws_resources import Stack
+from common.boto3.common import AWSClientError
+from common.imagebuilder_utils import ImageBuilderStatusMapping
+from pcluster.models.common import BaseTag
+from pcluster.schemas.imagebuilder_schema import ImageBuilderSchema
+from pcluster.templates.cdk_builder import CDKTemplateBuilder
+from pcluster.utils import get_installed_version
+
+LOGGER = logging.getLogger(__name__)
+
+
+class ImageBuilderActionError(Exception):
+    """Represent an error during the execution of an action on the imagebuilder."""
+
+    def __init__(self, message: str, validation_failures: list = None):
+        super().__init__(message)
+        self.validation_failures = validation_failures or []
+
+
+class ImageBuilderStack(Stack):
+    """Class representing a running stack associated to a building image."""
+
+    def __init__(self, name: str):
+        """Init stack info."""
+        super().__init__(name)
+        try:
+            self._image_resource = AWSApi().instance().cfn.describe_stack_resouce(name, "ParallelClusterImage")
+        except AWSClientError:
+            self._image_resource = None
+
+    @property
+    def version(self):
+        """Return the version of ParallelCluster used to create the stack."""
+        return self._get_tag("pcluster_version")
+
+    @property
+    def imagebuild_status(self):
+        """Return the status of the stack of build image process."""
+        cfn_status = self._stack_data.get("StackStatus")
+        for key, value in ImageBuilderStatusMapping.items():
+            if cfn_status in value:
+                return key
+        return None
+
+    @property
+    def image_id(self):
+        """Return output image id."""
+        if self._image_resource:
+            image_build_version_arn = self._image_resource.get("StackResourceDetail").get("PhysicalResourceId")
```

Comment:
I think you can create an ImageInfo AWS resources (take a look at InstanceInfo in the https://github.com/aws/aws-parallelcluster/pull/2483 ) that give you all the attribute you need.